packages/app/src/providers/models.ts

```diff
@@ -1,5 +1,7 @@
+import type { ArrayShape, Dataset, ScalarShape } from '@h5web/shared';
+
 export interface ValuesStoreParams {
-  path: string;
+  dataset: Dataset<ScalarShape | ArrayShape>;
```

Comment:
Note that I'm actually enforcing that `valuesStore.getValue()` never gets called with a dataset with a null shape, which is another benefit.